src/summaries/smart_contracts.yaml

```diff
@@ -0,0 +1,159 @@
+      * A single transaction starts with an end user from an EOA, but it can pass through many dApps (decentralised Applications) before completing.
+      * Transaction begins with a single smart contract and continues to carry on all the intermediate steps within the contract body.
+    
+    - Atomicity
+      * This is one of the most important property of Ethereum Transaction Mechanism which makes DeFi different from CeFi.
+      * The transactions in Ethereum are atomic which means the clauses in a smart contract can lead to failure of transaction and thus all the steps of the transaction are reversed.
+      * Atomicity is a critical feature of transactions because funds can
+      move between many contracts (i.e., “exchange hands”) with the
+      knowledge and security that if one of the conditions is not met,
+      the contract terms reset as if the money never left the starting
+      point.
+
+- title: Miner Extractable Value 
+  shortTitle: MEV
+  key: miner-extractable-value
+  details: |
+    - Mempool
+      * Transactions are posted to a memory pool, or mempool, before
+      they are added to a block. 
+      * Miners monitor these posted transactions, add them to their own
+      mempool, and share the transaction with other miners to be
+      included in the next available block.
+    
+    - Miner Extractable Value
+      * Any actor can see transactions in the mempool by running or
+      communicating with mining nodes.
+      * This observation allows for front-running (legal in the DeFi space) to miners. If miner encounters a large order, they can plave their order before that, hence the miners are incentivised to do so.
+      * Miner extractable value (MEV) is a measure of the profit that the
+      miner could make by including, excluding or re-ordering
+      transactions. 
+      * This is one of the drawback of the Proof of work consensus mechanism. Obfuscating transactions can mitigate MEV, thus hiding from miners how they might profit from transactions.
+
+- title: Smart Contracts, Gas and Oracles  
+  shortTitle: Ethereum Components
+  key: smartcontracts-gas-oracles
+  details: |
+    - Smart Contracts
+      * Ethereum is the example of most popular smart contract platform.
+      * A smart contract is code that can create and transform arbitrary data or
+      tokens on top of the blockchain of which it is a part.
+      * These are extremely powerful because it allows users to trustlessly encode
+      rules for any type of transaction and even create scarce assets with
+      specialized functionality.
+      * Many standard business contracts can be algorithmically encoded and
+      algorithmically enforced. These contracts run on the Ethereum blockchain and are run on every node.
+      * This is not just useful for finance. For example, supply chains are another
+      good example.
+
+    - Gas 
+      * Transactions have a gas fee that is the fee received by the miners for conducting transactions.
```

Comment:
We should also explain about mining as it goes hand in hand with Gas etc.

Reply:
Done